crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -1870,53 +1921,12 @@ async fn value_visitor_inner(
             _,
             box JsValue::WellKnownFunction(WellKnownFunctionKind::RequireResolve),
             args,
-        ) => {
-            if args.len() == 1 {
-                let pat = js_value_to_pattern(&args[0]);
-                let request = RequestVc::parse(Value::new(pat.clone()));
-                let resolved = cjs_resolve(
-                    origin,
-                    request,
-                    OptionIssueSourceVc::none(),
-                    try_to_severity(in_try),
-                )
-                .await?;
-                let mut values = resolved
-                    .primary
-                    .iter()
-                    .map(|result| async move {
-                        Ok(if let PrimaryResolveResult::Asset(asset) = result {
-                            Some(require_resolve(asset.ident().path()).await?)
-                        } else {
-                            None
-                        })
-                    })
-                    .try_join()
-                    .await?
-                    .into_iter()
-                    .flatten()
-                    .collect::<Vec<_>>();
-                match values.len() {
-                    0 => JsValue::Unknown(
-                        Some(Arc::new(JsValue::call(
-                            box JsValue::WellKnownFunction(WellKnownFunctionKind::RequireResolve),
-                            args,
-                        ))),
-                        "unresolveable request",
-                    ),
-                    1 => values.pop().unwrap(),
-                    _ => JsValue::alternatives(values),
-                }
-            } else {
-                JsValue::Unknown(
-                    Some(Arc::new(JsValue::call(
-                        box JsValue::WellKnownFunction(WellKnownFunctionKind::RequireResolve),
-                        args,
-                    ))),
-                    "only a single argument is supported",
-                )
-            }
-        }
+        ) => require_resolve_visitor(origin, args, in_try).await?,
+        JsValue::Call(
+            _,
+            box JsValue::WellKnownFunction(WellKnownFunctionKind::RequireContext),
+            args,
+        ) => require_context_visitor(origin, args, in_try).await?,
```

Comment:
static evaluation for require.context... You are really (over)killing it...

webpack doesn't even support that.

So I can do something fancy like:

```js
const context = require.context("...");
const resolved = context.resolve("./a");
const imported = require(resolved); // <- usually this would be too dynamic
```

Reply:
yea, the static evaluation was the reason I didn't put the context into a separate asset/module